src/base/static/components/input-form/form-category-menu-wrapper.js

```diff
@@ -22,26 +23,36 @@ class FormCategoryMenuWrapper extends Component {
       selectedCategory:
         this.visibleCategoryConfigs.length === 1
           ? this.visibleCategoryConfigs[0].category
-          : false,
+          : null,
+      isCategoryBarHidden: this.visibleCategoryConfigs.length === 1,
+      isSingleCategory: this.visibleCategoryConfigs.length === 1,
     };
   }
 
   onCategoryChange(selectedCategory) {
     this.setState({
       selectedCategory: selectedCategory,
+      isCategoryBarHidden: !!getCategoryConfig(selectedCategory)
+        .hide_category_button_on_selection,
```

Comment:
Does this need to be its own flag? Instead of having a separate `hide_category_button_on_selection` flag, it seems simpler to hide the Category Selector on select when a multi-stage form is configured. ie: `!!getCategoryConfig(selectedCategory).multi_stage`.

Also, is `isCategoryBarHidden` always going to be set to `hide_category_button_on_selection`, regardless of whether `this.visibleCategoryConfigs.length === 1`, like it is in the constructor? Eg: If there is only one visible category config, but the form is not configured to hide the Category Selector on select, shouldn't we still be hiding it?

Reply:
Originally I was imagining that we'd want the option of hiding or showing the selected category bar in the multi-stage workflow, but I think you're right-- this is probably not a useful option. It would be better to always hide the category button when entering a multi-stage workflow.